### PR TITLE
fix double click connection to delete for qt gui

### PR DESCRIPTION
**Pull request by ZakyHermawan**
## Description

Fix AttributeError while deleting connection with double click

## Related Issue

Fixes https://github.com/gnuradio/gnuradio/issues/7269

## Which blocks/areas does this affect?

`grc/gui_qt/components/canvas/connection.py`

## Testing Done

- Manual gui test
- Automated gui testing with `pytest grc/tests/test_qtbot.py`

## Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [x] I have squashed my commits to have one significant change per commit. 
- [x] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [x] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [x] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [x] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

File: grc/gui_qt/components/canvas/connection.py [hunk 1]
```diff
@@ -159,5 +159,6 @@ def paint(self, painter, option, widget):
         painter.drawPath(self._arrowhead)
 
     def mouseDoubleClickEvent(self, e):
-        self.parent.connections.remove(self)
+        connection = self.core
+        connection.source_port.parent_flowgraph.connections.remove(connection)
```
Comment (willcode) [inline grc/gui_qt/components/canvas/connection.py:163]: Isn't there a `connection.parent`? Or did that not work?

Comment (ZakyHermawan) [reply grc/gui_qt/components/canvas/connection.py]: you're right. I think using `connection.parent` is better than `connection.source_port.parent_flowgraph`

**Comment by willcode** [comment]
Thanks. For future PRs, please start the commit message (and PR title) with the module name (and block if applicable), like
`grc-qt: fix thing`
This make it easier to run through the commits and PRs later.

**Comment by willcode** [comment]
@ZakyHermawan see #7284 ... there should be a better way to do this, e.g., delete element. I'm going to hold off backporting this because it leaves the flowgraph in an invalid state but doesn't mark it as such. Take a look at how the select/delete way of removing a connection works.

**Pull request merged**